nav2_behavior_tree/plugins/action/assisted_teleop_action.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+
+#include "nav2_behavior_tree/plugins/action/assisted_teleop_action.hpp"
+
+namespace nav2_behavior_tree
+{
+
+AssistedTeleopAction::AssistedTeleopAction(
+  const std::string & xml_tag_name,
+  const std::string & action_name,
+  const BT::NodeConfiguration & conf)
+: BtActionNode<nav2_msgs::action::AssistedTeleop>(xml_tag_name, action_name, conf)
+{
+  double time_allowance;
+  getInput("time_allowance", time_allowance);
+  getInput("recovery", recovery_);
+
+  // Populate the input message
+  goal_.time_allowance = rclcpp::Duration::from_seconds(time_allowance);
+}
+
+void AssistedTeleopAction::on_tick()
+{
+  if (recovery_) {
+    increment_recovery_count();
+  }
+}
+
```

Comment:
Also we never use `is_recovery_` with the result to set the status of the BT node based on if the failure due to timeout means that the BT node should return a failure or a success (if not a recovery; that its just done and always interpret completion as success).